redwood-treehouse-guest/src/commonMain/kotlin/app/cash/redwood/treehouse/treehouseCompose.kt

```diff
@@ -55,6 +58,11 @@ private class RedwoodZiplineTreehouseUi(
    */
   override val scope = (treehouseUi as? ZiplineScoped)?.scope ?: ZiplineScope()
 
+  private val coroutineScope = CoroutineScope(
+    appLifecycle.coroutineScope.coroutineContext +
+      appLifecycle.coroutineScope.coroutineContext.job,
+  )
```

Comment:
```suggestion
  private val coroutineScope = CoroutineScope(
    appLifecycle.coroutineScope.coroutineContext +
      Job(appLifecycle.coroutineScope.coroutineContext.job),
  )
```
I think you probably still want to put in a new `Job` which gets canceled. Or does creating a new `CoroutineScope` do that automatically? If so, you probably don't need the `+ ...` part.

Reply:
Oh woops, that's right. A new Job is only created if there's no Job in the context. This scope's job should be a child of the parent scope's job, but this is giving both scopes the same job. Will fix!